realm/src/androidTest/java/io/realm/internal/JNITransactions.java

```diff
@@ -376,4 +378,13 @@ public void testAddEmptyRowWithPrimaryKeyLong() {
         assertEquals(1, t.size());
         assertEquals(42, t.getRow(rowIndex).getLong(0));
     }
+
+    public void testPrimaryKeyTableMigration() throws IOException {
```

Comment:
Is this test sufficient? Look at the migration code and consider how you ensure that changing a single line of code there would get checked in this test.
- I can't see that any of the row copying is being tested.
- What if the Primary key column was at various different indexes (front, middle, end)?